Bug in filter and alias

### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.


### Issue description

alias do not work after filter in edge case 

### Reproducible example

```python
import polars as pl
df = (
    pl.DataFrame({'A' : [1,2,3]}).lazy()
    .filter(pl.col('A') >= 3)  # seems to work with other numbers
    .with_column(-pl.col('A').alias('B'))
    .collect()
)
assert df.columns == ['A', 'B'], df.columns
# AssertionError: ['A', 'literal']
```


### Expected behavior

The column 'literal' should be named 'B'

### Installed versions

<details>

```
---Version info---
Polars: 0.15.1
Index type: UInt32
Platform: Linux-3.10.0-862.el7.x86_64-x86_64-with-glibc2.10
Python: 3.8.8 | packaged by conda-forge | (default, Feb 20 2021, 16:22:27) 
[GCC 9.3.0]
---Optional dependencies---
pyarrow: 2.0.0
pandas: 1.4.2
numpy: 1.20.2
fsspec: 2021.04.0
connectorx: <not installed>
xlsx2csv: <not installed>
matplotlib: 3.4.1
```

</details>

The reason here is the negative sign causing an order of operations, but it's weird it doesn't happen without a filter in there. What's happening:

1. You filter.
2. You select column `A` and alias it to `B`.
3. You negative column `B`.

The alias doesn't make it through because you applied an operation after the aliasing. With some parentheses, your problem can be fixed:

```
import polars as pl
df = (
    pl.DataFrame({'A' : [1,2,3]}).lazy()
    .filter(pl.col('A') >= 3)  # seems to work with other numbers
    .with_column((-pl.col('A')).alias('B')) # the alias is the final operation
    .collect()
)
assert df.columns == ['A', 'B'], df.columns
# AssertionError: ['A', 'literal']
```

There's probably a bug in here somewhere though, considering the behavior changes if you remove the `.filter` step.

--- a/polars/polars-core/src/chunked_array/arithmetic.rs
+++ b/polars/polars-core/src/chunked_array/arithmetic.rs
@@ -136,8 +136,9 @@ where
             let opt_lhs = lhs.get(0);
             match opt_lhs {
                 None => ChunkedArray::full_null(lhs.name(), rhs.len()),
-                Some(lhs) => {
-                    rhs.apply_mut(|rhs| operation(lhs, rhs));
+                Some(lhs_val) => {
+                    rhs.apply_mut(|rhs| operation(lhs_val, rhs));
+                    rhs.rename(lhs.name());
                     rhs
                 }
             }

--- a/polars/polars-lazy/src/tests/queries.rs
+++ b/polars/polars-lazy/src/tests/queries.rs
@@ -425,7 +425,7 @@ fn test_lazy_query_9() -> PolarsResult<()> {
 ))]
 fn test_lazy_query_10() {
     use polars_core::export::chrono::Duration as ChronoDuration;
-    let date = NaiveDate::from_ymd(2021, 3, 5);
+    let date = NaiveDate::from_ymd_opt(2021, 3, 5).unwrap();
     let x: Series = DatetimeChunked::from_naive_datetime(
         "x",
         [
@@ -502,7 +502,7 @@ fn test_lazy_query_10() {
     feature = "dtype-datetime"
 ))]
 fn test_lazy_query_7() {
-    let date = NaiveDate::from_ymd(2021, 3, 5);
+    let date = NaiveDate::from_ymd_opt(2021, 3, 5).unwrap();
     let dates = [
         NaiveDateTime::new(date, NaiveTime::from_hms_opt(12, 0, 0).unwrap()),
         NaiveDateTime::new(date, NaiveTime::from_hms_opt(12, 1, 0).unwrap()),

--- a/py-polars/tests/unit/test_schema.py
+++ b/py-polars/tests/unit/test_schema.py
@@ -267,3 +267,14 @@ def test_boolean_agg_schema() -> None:
             == agg_df.schema
             == {"x": pl.Int64, "max_y": pl.Boolean}
         )
+
+
+def test_schema_owned_arithmetic_5669() -> None:
+    df = (
+        pl.DataFrame({"A": [1, 2, 3]})
+        .lazy()
+        .filter(pl.col("A") >= 3)
+        .with_column(-pl.col("A").alias("B"))
+        .collect()
+    )
+    assert df.columns == ["A", "literal"], df.columns

